include/exadg/operators/inverse_mass_operator.h

```diff
@@ -315,14 +346,59 @@ class InverseMassOperator
     Integrator                      integrator(*matrix_free, dof_index, quad_index);
     InverseMassAsMatrixFreeOperator inverse_mass(integrator);
 
-    for(unsigned int cell = cell_range.first; cell < cell_range.second; ++cell)
+    if(coefficient_is_variable)
     {
-      integrator.reinit(cell);
-      integrator.read_dof_values(src, 0);
+      for(unsigned int cell = cell_range.first; cell < cell_range.second; ++cell)
+      {
+        integrator.reinit(cell);
+        integrator.read_dof_values(src, 0);
 
-      inverse_mass.apply(integrator.begin_dof_values(), integrator.begin_dof_values());
+        dealii::AlignedVector<dealii::VectorizedArray<Number>> inverse_JxW_times_coefficient(
+          this->matrix_free->get_dofs_per_cell(dof_index));
+        inverse_mass.fill_inverse_JxW_values(inverse_JxW_times_coefficient);
+
+        if(consider_inverse_coefficient)
```

Comment:
We could move that `if(consider_inverse_coefficient)` out of the cell loop by duplicating a few lines of code.
Since the parameter `consider_inverse_coefficient` is typically constant per object (or at least per call of `cell_loop_matrix_free_operator`), I think the branch prediction is probably handling this well enough.